src/@chakra-ui/gatsby-plugin/components/Switch.ts

```diff
@@ -0,0 +1,66 @@
+import { createMultiStyleConfigHelpers } from "@chakra-ui/react"
+import { switchAnatomy } from "@chakra-ui/anatomy"
+import {
+  commonInputTriggerStyles,
+  defineMergeStyles,
+  switchDefaultTheme,
+  _notDisabledReadOnly,
+} from "./components.utils"
+
+const { defineMultiStyleConfig, definePartsStyle } =
+  createMultiStyleConfigHelpers(switchAnatomy.keys)
+
+const { baseStyle: defaultBaseStyle, sizes: defaultSizes } = switchDefaultTheme
+
+const switchSmSize = defaultSizes?.sm
+
+const { commonControlProps, commonContainerProps, commonLabelProps } =
+  commonInputTriggerStyles
+
+const baseStyleContainer = defineMergeStyles(
+  defaultBaseStyle?.({} as never).container,
+  switchSmSize?.container,
+  commonContainerProps
+)
+
+const baseStyleThumb = defineMergeStyles(
+  defaultBaseStyle?.({} as never).thumb,
+  {
+    bg: "background.base",
+    [".chakra-switch__track:not([data-checked])[data-disabled] > &"]: {
+      border: "1px",
+      borderColor: "disabled",
+    },
+  }
+)
+
+const baseStyleTrack = defineMergeStyles(
+  defaultBaseStyle?.({} as never).track,
+  commonControlProps,
+  {
+    bg: "body.medium",
+    borderColor: "transparent",
+    _invalid: { borderColor: "error.outline" },
+    _focusVisible: { bg: "gray.500", borderColor: "transparent" },
+    ["&:not([data-checked])[data-disabled]"]: {
+      bg: "transparent",
+      borderColor: "disabled",
+    },
+    [_notDisabledReadOnly]: {
+      "*[data-checked]:hover > &, *:not([data-checked]):hover > &": {
```

Comment:
@TylerAPfledderer I had to do this to catch the two possible states and override the common styles. The switch is slightly different in terms of hover styles than the radio and checkbox (check DS).